lib/Core/Executor.cpp

```diff
@@ -3775,7 +3776,8 @@ void Executor::runFunctionAsMain(Function *f,
     }
   }
 
-  ExecutionState *state = new ExecutionState(kmodule->functionMap[f]);
+  auto state = new ExecutionState(kmodule->functionMap[f]);
+  state->setID();
```

Comment:
This is just unnecessary clutter. Is it really that bad if some ids are skipped due to allocations outside of these cases?

Reply:
It just initialises the first state to 1. I could probably also use the ctor here as this one is only used for the initial state.

Reply:
Moved it to ctor...